include/lttng/userspace-probe-internal.h

```diff
@@ -0,0 +1,143 @@
+	enum lttng_userspace_probe_location_lookup_method_type type;
+};
+
+struct lttng_userspace_probe_location_lookup_method_elf {
+	struct lttng_userspace_probe_location_lookup_method parent;
+};
+
+struct lttng_userspace_probe_location_lookup_method_sdt {
+	struct lttng_userspace_probe_location_lookup_method parent;
+};
+
+struct lttng_userspace_probe_location_comm {
+	/* enum lttng_userspace_probe_location_type */
+	int8_t type;
+	/*
+	 * Payload is composed of, in that order,
+	 *   - type-specific payload
+	 *   - struct lttng_userspace_probe_location_lookup_method_comm
+	 */
+	char payload[];
+};
+
+struct lttng_userspace_probe_location_function_comm {
+	/* Both lengths include the trailing \0. */
+	uint32_t function_name_len;
+	uint32_t binary_path_len;
+	/*
+	 * Payload is composed of, in that order,
+	 *   - function name (with trailing \0),
+	 *   - absolute binary path (with trailing \0)
+	 */
+	char payload[];
+} LTTNG_PACKED;
+
+struct lttng_userspace_probe_location_tracepoint_comm {
+	/* Both lengths include the trailing \0. */
+	uint32_t probe_name_len;
+	uint32_t provider_name_len;
+	uint32_t binary_path_len;
+	/*
+	 * Payload is composed of, in that order,
+	 *   - probe name (with trailing \0),
+	 *   - provider name (with trailing \0),
+	 *   - absolute binary path (with trailing \0)
+	 */
+	char payload[];
+} LTTNG_PACKED;
+
+/* Common ancestor of all probe locations. */
+struct lttng_userspace_probe_location {
+	enum lttng_userspace_probe_location_type type;
+	struct lttng_userspace_probe_location_lookup_method *lookup_method;
+};
+
+struct lttng_userspace_probe_location_function {
+	struct lttng_userspace_probe_location parent;
+	char *function_name;
+	char *binary_path;
+	/* Set to -1 if not open. */
+	int binary_fd;
+};
+
+struct lttng_userspace_probe_location_tracepoint {
+	struct lttng_userspace_probe_location parent;
+	char *probe_name;
+	char *provider_name;
+	char *binary_path;
+	/* Set to -1 if not open. */
+	int binary_fd;
+};
+
+LTTNG_HIDDEN
+int lttng_userspace_probe_location_serialize(
+		struct lttng_userspace_probe_location *location,
+		struct lttng_dynamic_buffer *buffer,
+		int *binary_fd);
+
+LTTNG_HIDDEN
+int lttng_userspace_probe_location_create_from_buffer(
+		const struct lttng_buffer_view *buffer,
+		struct lttng_userspace_probe_location **probe_location);
+
+LTTNG_HIDDEN
+int lttng_userspace_probe_location_function_set_binary_fd(
+		struct lttng_userspace_probe_location *location, int binary_fd);
+
+LTTNG_HIDDEN
+int lttng_userspace_probe_location_tracepoint_set_binary_fd(
+		struct lttng_userspace_probe_location *location, int binary_fd);
+
+/*
+ * Returns a version of the location that is serialized to a contiguous region
+ * of memory. Pass NULL to buffer to only get the storage requirement of the
+ * flattened userspace probe location.
+ */
+LTTNG_HIDDEN
+int lttng_userspace_probe_location_flatten(
+		struct lttng_userspace_probe_location *location,
```

Comment:
`struct lttng_userspace_probe_location *location` can be marked as `const`.
`const struct lttng_userspace_probe_location *location`

Reply:
fixed